packages/fuels-contract/src/script.rs

```diff
@@ -402,4 +463,344 @@ mod test {
+                        coin.asset_id.into(),
+                        0,
+                        0,
+                    )
+                })
+                .collect::<HashSet<_>>();
+
+            assert_eq!(expected_inputs, inputs_as_signed_coins);
+        }
+
+        #[test]
+        fn variable_outputs_appended_to_outputs() {
+            // given
+            let variable_outputs = [100, 200].map(|amount| {
+                Output::variable(random_bech32_addr().into(), amount, Default::default())
+            });
+
+            let calls = variable_outputs
+                .iter()
+                .cloned()
+                .map(|variable_output| {
+                    given_a_contract_call().with_variable_outputs(vec![variable_output])
+                })
+                .collect::<Vec<_>>();
+
+            // when
+            let (_, outputs) = Script::get_transaction_inputs_outputs(
+                &calls,
+                &random_bech32_addr(),
+                Default::default(),
+            );
+
+            // then
+            let actual_variable_outputs: HashSet<Output> = outputs[2..].iter().cloned().collect();
+            let expected_outputs: HashSet<Output> = variable_outputs.into();
+
+            assert_eq!(expected_outputs, actual_variable_outputs);
+        }
+    }
+
+    fn given_a_contract_call() -> ContractCall {
+        ContractCall {
+            contract_id: random_bech32_contract_id(),
+            encoded_args: Default::default(),
+            encoded_selector: [0; 8],
+            call_parameters: Default::default(),
+            compute_custom_input_offset: false,
+            variable_outputs: None,
+            external_contracts: Default::default(),
+            output_param: None,
+        }
+    }
+
+    impl ContractCall {
+        pub fn with_contract_id(self, contract_id: Bech32ContractId) -> Self {
+            ContractCall {
+                contract_id,
+                ..self
+            }
+        }
+        pub fn with_external_contracts(
+            self,
+            external_contracts: Vec<Bech32ContractId>,
+        ) -> ContractCall {
+            ContractCall {
+                external_contracts,
+                ..self
+            }
+        }
+
+        pub fn with_variable_outputs(self, variable_outputs: Vec<Output>) -> ContractCall {
+            ContractCall {
+                variable_outputs: Some(variable_outputs),
+                ..self
+            }
+        }
+
+        pub fn with_call_parameters(self, call_parameters: CallParameters) -> ContractCall {
+            ContractCall {
+                call_parameters,
+                ..self
+            }
+        }
+    }
+
+    fn random_bech32_addr() -> Bech32Address {
+        Bech32Address::new("fuel", rand::thread_rng().gen::<[u8; 32]>())
+    }
+
+    fn random_bech32_contract_id() -> Bech32ContractId {
+        Bech32ContractId::new("fuel", rand::thread_rng().gen::<[u8; 32]>())
+    }
```

Comment:
These can be `impl` methods of the types.

Reply:
Great idea! I'll do that now.

Reply:
Unfortunately, this only works if you impl your own trait or if the type is in your crate. 

I'd rather not add a rand dependency to the users of bech32 until a real-life need presents itself.